packages/elements/src/components/ino-datepicker/validation.ts

```diff
@@ -3,58 +3,191 @@ import flatpickr from 'flatpickr';
 
 const NUMBERS_WITH_SPECIAL_CHARS = /(\d[^a-z]+)/g;
 
-export const validateRange = (value: string, format: string): boolean =>
-  !value
-    .match(NUMBERS_WITH_SPECIAL_CHARS)
-    .map((match) => validateFormat(match.trim(), format))
-    .includes(false);
-
-export const validateSingle = (
-  value: string,
-  format: string,
-  minDateStr?: string,
-  maxDateStr?: string
-): boolean =>
-  validateMin(value, format, minDateStr) &&
-  validateMax(value, format, maxDateStr) &&
-  validateFormat(value, format);
-
-export const validateMin = (
-  value: string,
-  format: string,
-  minDateStr?: string
-): boolean => {
-  if (!minDateStr) {
-    return true;
-  }
-  const parsedDate: Date = parseDate(value, format);
-  const minDate = flatpickr.parseDate(minDateStr, format);
-  return isEqual(parsedDate, minDate) || isAfter(parsedDate, minDate);
+type Props = {
+  disabled: boolean;
+  required: boolean;
+  dateFormat: string;
+  isRanged: boolean;
+  minDate?: string;
+  maxDate?: string;
 };
 
-export const validateMax = (
-  value: string,
-  format: string,
-  maxDateStr?: string
-): boolean => {
-  if (!maxDateStr) {
+export class DateValidator {
```

Comment:
Maybe Validator as class name is more universal. `Date` only relates to the type Dates. e.g. datetime, month as well as time are not covered